test/unit/overlay.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+var fixtures = require('../fixtures');
+var fs = require('fs');
+var sharp = require('../../index');
+
+sharp.cache(0);
+
+// Main
+describe('Overlays', function() {
+  it('Overlay transparent PNG on solid background', function(done) {
+    sharp(fixtures.inputPngOverlayLayer0)
+      .overlayWith(fixtures.inputPngOverlayLayer1)
+      .toBuffer(function (error, data, info) {
+        if (error) return done(error);
+
+        fixtures.assertSimilar(fixtures.expected('alpha-layer-01.png'), data, done);
+      });
+  });
+
+  it('Composite three transparent PNGs into one', function(done) {
+    sharp(fixtures.inputPngOverlayLayer0)
+      .overlayWith(fixtures.inputPngOverlayLayer1)
+      .toBuffer(function (error, data, info) {
+        if (error) return done(error);
+
+        sharp(data)
+          .overlayWith(fixtures.inputPngOverlayLayer2)
+          .toBuffer(function (error, data, info) {
+            if (error) return done(error);
+
+            // TODO: Remove once this fails
+            fixtures.assertSimilar(fixtures.expected('alpha-layer-012-low-alpha.png'), data, function (error, result) {
```

Comment:
Thanks, I added `threshold` option 23a151f20219a42d0615dfe0c851aee24607afae and it seems to have worked detecting that case using `threshold: 0`. However, I still wonder if it’d be valuable to have a identity check at an image level (ignoring metadata, etc) for testing subtle changes such as sharpening, opacity, etc.